### PR TITLE
Port Klein to Python 3

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by hawkowl** [comment]
This requires https://twistedmatrix.com/trac/ticket/7996 and https://twistedmatrix.com/trac/ticket/7994 .

**Comment by glyph** [comment]
#86

**Comment by hawkowl** [comment]
All tests are green. cc @glyph

File: klein/app.py [hunk 1]
```diff
@@ -78,6 +81,8 @@ def __ne__(self, other):
         return not result
 
 
+
```
Comment (glyph) [inline klein/app.py]: This whitespace seems excessive?

File: klein/test/test_resource.py [hunk 2]
```diff
@@ -101,7 +104,7 @@ def getWrittenData():
 def _render(resource, request, notifyFinish=True):
     result = resource.render(request)
 
-    if isinstance(result, str):
+    if isinstance(result, bytes):
```
Comment (glyph) [inline klein/test/test_resource.py]: I think we probably also need to be able to handle text being returned by rendering as well?

Comment (glyph) [inline klein/test/test_resource.py]: Actually nevermind; this is probably good enough for test code.

File: klein/test/test_resource.py [hunk 3]
```diff
@@ -1014,7 +1017,19 @@ def test_urlDecodeErrorRepr(self):
             repr(_URLDecodeError(ValueError)),
         )
 
+    def test_urlDecodeErrorReprPy3(self):
+        """
+        URLDecodeError.__repr__ formats properly.
```
Comment (glyph) [inline klein/test/test_resource.py]: "properly" is one of those words we should avoid: https://jml.io/pages/test-docstrings.html

**Comment by glyph** [comment]
This is a shockingly small change, and my feedback is all nitpicky (the feedback is really "we need a tox lint job"); I look forward to a Twisted release with the relevant bits ported so that we can move this out of `allowed_failures`.

**Pull request merged**